### PR TITLE
Support for "tiny" guests

### PR DESCRIPTION
**Pull request by notmyname**
This enables the "tiny" test target for the dev test target use case. More specifically, this branch merges in:
 - tiny and basic saio manifest examples
 - appropriate config files for a tiny setup
 - more generic guest profile template with dynamic number of drives
 - support for a dynamic number of drives in the guest
 - rewrite of some shell to bash

### CODE REVIEW

File: start.py [hunk 1]
```diff
@@ -90,13 +91,14 @@ def get_manifest(workspace_name):
 
     container_name = workspace_name
 
+    vol_count = int(manifest.runway_options['number_of_drives'])
```
Comment (gerardgine) [inline start.py]: This will fail if 'number_of_drives' is not specified in the manifest.

File: simple_start.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+if __name__ == '__main__':
+    usage = '%s manifest_file' % sys.argv[0]
+
+    try:
+        manifest_path = os.path.abspath(sys.argv[1])
+    except IndexError:
+        print('Usage: %s' % usage)
+        sys.exit(1)
+
```
Comment (gerardgine) [inline simple_start.py]: Doesn't this script need to be run as root?

+ lxc profile edit swift-runway-tiny-001-profile
  WARNING: Running as a non-root user. Functionality may be unavailable.
  /run/lvm/lvmetad.socket: connect failed: Permission denied
  WARNING: Failed to connect to lvmetad. Falling back to internal scanning.
  /dev/mapper/control: open failed: Permission denied
  Failure to communicate with kernel device-mapper driver.
  Incompatible libdevmapper (unknown version) and kernel driver (unknown version).
  striped: Required device-mapper target(s) not detected in your kernel.
  Run `lvcreate --help' for more information.
  WARNING: Running as a non-root user. Functionality may be unavailable.
  /run/lvm/lvmetad.socket: connect failed: Permission denied
  WARNING: Failed to connect to lvmetad. Falling back to internal scanning.
  /run/lock/lvm/V_swift-runway-vg01:aux: open failed: Permission denied
  Can't get lock for swift-runway-vg01
  Cannot process volume group swift-runway-vg01

Comment (gerardgine) [inline simple_start.py]: The above output is some of the output I get when running it as an unprivileged user.

Comment (gerardgine) [inline simple_start.py]: So I guess we should do something in line of what we do in start.py (and used to do in the original start.sh):

```    if os.geteuid() != 0:
        exit_with_error("This script must be run as root.")
```

Comment (gerardgine) [inline simple_start.py]: When running as an unprivileged user, it all ends up failing with:

```
TASK [bootstrap : Make filesystems on disks] ***********************************
failed: [localhost] (item={u'mount_point': u'/srv/1/node/d1', u'drive_letter': u'sdb'}) => {"failed": true, "item": {"drive_letter": "sdb", "mount_point": "/srv/1/node/d1"}, "msg": "Device /dev/sdb not found."}
	to retry, use: --limit @/root/ansible/bootstrap.retry
```

So yes, I guess we must enforce being root.

File: simple_start.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    # create workspace dir from runway_options['family']
+    workspaces.WORKSPACE_PREFIX += '%s-' % man.runway_options['family']
+    workspace_name = workspaces.get_new_workspace_name()
+    workspace_path = workspaces.create_workspace_dir(workspace_name)
+    man.workspace_dir = workspace_path
+
+    # get manifest components into workspace
+    man.retrieve_components()
+
+    # get a copy fo the manifest into the workspace directory
```
Comment (gerardgine) [inline simple_start.py]: Minor typo in comment fo => of

**Comment by gerardgine** [comment]
In general, I'm very tempted to use some of the option you have in the manifest in the old/big start.py, but I'm not sure they belong in there (at least not all of them). The debug option, for instance, is a pure "run" option (a command line thing). I totally understand why you're doing this way in the tiny version, and I guess that's ok, but I guess start.py is fine the way it is.

The other big difference is that in the tiny version we create the workspace in behalf of the user, while we decided we, in general, wanted the workspace to already exist (and that's actually a feature that ProxyFS devs use all the time).

Besides that, our (Proxyfs) workflow still works, so I guess we're good to go. Will you merge it yourself?

File: start.py (hunk 1 above)
Comment (notmyname) [inline start.py]: done

File: simple_start.py (hunk 2 above)
Comment (notmyname) [inline simple_start.py]: yep. done

File: simple_start.py (hunk 3 above)
Comment (notmyname) [inline simple_start.py]: done

**Pull request merged**